app/gui2/src/components/DocumentationPanel.vue

```diff
@@ -0,0 +1,277 @@
+    if (isTag) {
+      tags.push(section.Tag)
+    } else if (isExample) {
+      examples.push(section.Marked)
+    } else {
+      synopsis.push(section)
+    }
+  }
+  return { tags, synopsis, examples }
+}
+
+// === Lookup ===
+
+function getChildren(db: SuggestionDb, id: SuggestionId, kind: SuggestionKind): Docs[] {
+  if (!id) return []
+  const children = Array.from(db.parent.reverseLookup(id))
+  return children.reduce((acc: Docs[], id: SuggestionId) => {
+    if (db.get(id)?.kind === kind) {
+      const docs = lookupDocumentation(db, id)
+      acc.push(docs)
+    }
+    return acc
+  }, [])
+}
+
+function asFunctionDocs(docs: Docs[]): FunctionDocs[] {
+  return docs.flatMap((doc) => {
+    if ('Function' in doc) {
+      return [doc.Function]
+    } else {
+      console.error(`Unexpected docs type: ${docs}, expected Function`)
+      return []
+    }
+  })
+}
+
+function asTypeDocs(docs: Docs[]): TypeDocs[] {
+  return docs.flatMap((doc) => {
+    if ('Type' in doc) {
+      return [doc.Type]
+    } else {
+      console.error(`Unexpected docs type: ${docs}, expected Type`)
+      return []
+    }
+  })
+}
+
+type DocsHandle = (db: SuggestionDb, entry: SuggestionEntry, id: SuggestionId) => Docs
+
+const handleFunction: DocsHandle = (_db, entry, id) => ({
+  Function: {
+    name: entry.name,
+    id,
+    arguments: entry.arguments,
+    sections: filterSections(entry.documentation),
+  },
+})
+
+const handleDocumentation: Record<SuggestionKind, DocsHandle> = {
+  [SuggestionKind.Function]: handleFunction,
+  [SuggestionKind.Method]: handleFunction,
+  [SuggestionKind.Constructor]: handleFunction,
+  [SuggestionKind.Local]: (_db, entry, id) => ({
+    Local: {
+      name: entry.name,
+      id,
+      sections: filterSections(entry.documentation),
+    },
+  }),
+  [SuggestionKind.Type]: (db, entry, id) => {
+    return {
+      Type: {
+        name: entry.name,
+        id,
+        arguments: entry.arguments,
+        sections: filterSections(entry.documentation),
+        methods: asFunctionDocs(getChildren(db, id, SuggestionKind.Method)),
+        constructors: asFunctionDocs(getChildren(db, id, SuggestionKind.Constructor)),
+      },
+    }
+  },
+  [SuggestionKind.Module]: (db, entry, id) => {
+    return {
+      Module: {
+        name: entry.name,
+        id,
+        sections: filterSections(entry.documentation),
+        types: asTypeDocs(getChildren(db, id, SuggestionKind.Type)),
+        methods: asFunctionDocs(getChildren(db, id, SuggestionKind.Method)),
+      },
+    }
+  },
+}
+
+function lookupDocumentation(db: SuggestionDb, id: SuggestionId): Docs {
+  const entry = db.get(id)
+  if (!entry) return { Placeholder: 'Entry not found' }
+  const handle = handleDocumentation[entry.kind]
+  return handle ? handle(db, entry, id) : { Placeholder: `Entry kind not handled: ${entry.kind}` }
+}
+
+// === Helper variables ===
+
+const documentation = computed<Docs>(() => {
+  const db = useSuggestionDbStore()
```

Comment:
The stores in general should be accessed at the module level, not inside a computed value.

Reply:
Fixed